### ISSUE TITLE
ClassCastException in DistinctValueAggregation.DistinctValueMapper.map() when gettings values from PredicateSupplier

### ISSUE THREAD

**Issue by spongyT** [hazelcast/hazelcast#7398]
Assume having the following Entities and I want to get a set of unique values for an attribute "type":

E1: {"assetId":"a", "type":"car"}
E2: {"assetId":"b", "type":"paper"}
E3: {"assetId":"c", "type":"blub"}
E3: {"assetId":"d", "type":"car"}

I just want to take a look into entities E1, E2 and E4 and the result should be a set containing "car" and "paper". Due to that I wrote the following code sinppet:

```
com.hazelcast.query.Predicate<String, SampleEntity> assetIdsRestriction = Predicates.in("assetId", "a", "b", "d");
Supplier<String, SampleEntity, String> typeExtractorSupplier = Supplier.all(entity -> entity.getType());
Supplier<String, FleetEntity, String> supplier = Supplier.fromPredicate(assetIdsRestriction, typeExtractorSupplier );
Aggregation<String, String, Set<String>> aggregation = Aggregations.distinctValues();
        return map.aggregate(supplier, aggregation);
```

And what I get is a ClassCastException in DistinctValueAggregation.DistinctValueMapper.map():
```
Caused by: java.lang.ClassCastException: com.hazelcast.mapreduce.aggregation.impl.DistinctValuesAggregation$SimpleEntry cannot be cast to com.hazelcast.query.impl.Extractable
	at com.hazelcast.query.impl.predicates.AbstractPredicate.readAttributeValue(AbstractPredicate.java:129)
	at com.hazelcast.query.impl.predicates.AbstractPredicate.apply(AbstractPredicate.java:55)
	at com.hazelcast.mapreduce.aggregation.impl.PredicateSupplier.apply(PredicateSupplier.java:58)
	at com.hazelcast.mapreduce.aggregation.impl.DistinctValuesAggregation$DistinctValueMapper.map(DistinctValuesAggregation.java:202)
	at com.hazelcast.mapreduce.impl.task.KeyValueSourceMappingPhase.executeMappingPhase(KeyValueSourceMappingPhase.java:49)
	at com.hazelcast.mapreduce.impl.task.MapCombineTask.processMapping(MapCombineTask.java:135)
	at com.hazelcast.mapreduce.impl.task.MapCombineTask.processPartitionMapping(MapCombineTask.java:304)
	at com.hazelcast.mapreduce.impl.task.MapCombineTask.access$700(MapCombineTask.java:68)
	at com.hazelcast.mapreduce.impl.task.MapCombineTask$PartitionBasedProcessor.processPartitions(MapCombineTask.java:356)
	at com.hazelcast.mapreduce.impl.task.MapCombineTask$PartitionBasedProcessor.run(MapCombineTask.java:333)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at com.hazelcast.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:67)
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:92)
```

This is because DistinctValueAggregation.DistinctValueMapper.SimpleEntry only implements Map.Entry<K, V> 
```
private static final class SimpleEntry<K, V> implements Map.Entry<K, V> { }
```

and doesn´t extend from QueryableEntry as done in SupplierConsumingMapper.SimpleEntry:

```
static final class SimpleEntry<K, V> extends QueryableEntry<K, V> 
```

---This only happens on Hazelcast 3.6-RC1.----

**Comment by jerrinot** [comment]
Hi @spongyT,

thank you for a nice bugreport. Let me see what I can do about it. The fix should be straightforward.

**Comment by adasari** [comment]
Hi ,
we are facing same issue in 3.5.2 as well .Was the fix part of 3.5.2 also ? or 3.6.1 onwards ? Thanks.
